File: 942 - DI String Match.cpp
class Solution {
public:
    vector<int> diStringMatch(string s) {
        int l = 0, r = s.size();
        vector<int> res;
        for (char c : s) {
            if (c == 'I') {
                res.push_back(l++);
            } else {
                res.push_back(r--);
            }
        }
        res.push_back(r);
        return res;
    }
};

File: 942 - DI String Match.js
/**
 * @param {string} s
 * @return {number[]}
 */
var diStringMatch = function (s) {
    let l = 0, r = s.length, i = 0;
    let res = [];
    for (let c of s) {
        if (c == 'I') {
            res.push(l++);
        } else {
            res.push(r--);
        }
    }
    res.push(l);
    return res;
};